### PR TITLE
Ng 43 code editor

### PR DESCRIPTION
**Pull request by alejoreyes96**
Implemented code editor functionality. When user loads file, text on file will be displayed on editor. User is able to run code on the editor, as well as save as a file whatever there is written. Added clear warning and fixed some bugs with run/debug that were interfering with the editor. Also added a menu for saving memory/register content and saving editor content.

### CODE REVIEW

**Comment by cfboy** [comment]
Add support for .obj files.

File: GUI/semref.kv [hunk 1]
```diff
@@ -219,6 +219,11 @@
             pos: dp(140), dp(140)
             size: dp(40), dp(10)
 
-<HexKeyboard>:
+<TextEditor>
+    multiline: True
+    size_hint: 0.50, 0.43
+    pos_hint: {'x' : 0.23, 'y': 0.05}
```
Comment (ChristianPerez34) [inline GUI/semref.kv]: Remove these

File: GUI/window.py [hunk 2]
```diff
@@ -1040,6 +1208,47 @@ def update_ascii_grid(self):
             i += 1
 
 
+class TextEditor(TextInput):
+
+    def __init__(self, **kwargs):
+        super(TextEditor, self).__init__(**kwargs)
+        self.bind(text=self.on_text)
+        self.valid_text = False
+        self.markup = True
```
Comment (ChristianPerez34) [inline GUI/window.py]: Remove `self.markup`
```Python
if self.dpi < 192:
            self.size_hint = (0.50, 0.43)
            self.pos_hint = {
                'x' : 0.23, 
                'y': 0.05
                }
        else:
            self.size_hint = (0.50, 0.43)
            self.pos_hint = {
                'x' : 0.24, 
                'y': 0.05
                }
```

**Review by ChristianPerez34 — CHANGES_REQUESTED** [review]

File: GUI/window.py (hunk 2 above)
Comment (alejoreyes96) [reply GUI/window.py]: you mean add this snippet?

Comment (ChristianPerez34) [reply GUI/window.py]: Yes

**Comment by ChristianPerez34** [comment]
.
Screen Shot 2019-11-12 at 10 09 34 AM

**Comment by alejoreyes96** [comment]
> .
> Screen Shot 2019-11-12 at 10 09 34 AM

I'll check it out.

**Review by ChristianPerez34 — APPROVED** [review]

**Review by cfboy — APPROVED** [review]

**Review by noelvalentin — APPROVED** [review]

**Pull request merged**